doc/api/esm.md

```diff
@@ -1816,12 +1844,12 @@ success!
 [`import`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import
 [`module.createRequire()`]: modules.html#modules_module_createrequire_filename
 [`module.syncBuiltinESMExports()`]: modules.html#modules_module_syncbuiltinesmexports
-[`transformSource` hook]: #esm_code_transformsource_code_hook
-[ArrayBuffer]: https://www.ecma-international.org/ecma-262/6.0/#sec-arraybuffer-constructor
-[SharedArrayBuffer]: https://tc39.es/ecma262/#sec-sharedarraybuffer-constructor
-[string]: https://www.ecma-international.org/ecma-262/6.0/#sec-string-constructor
-[TypedArray]: https://www.ecma-international.org/ecma-262/6.0/#sec-typedarray-objects
-[Uint8Array]: https://www.ecma-international.org/ecma-262/6.0/#sec-uint8array
+[`transformSource` hook]: #esm_code_transformsource_source_context_defaulttransformsource_code
+[`ArrayBuffer`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer
+[`SharedArrayBuffer`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer
+[`string`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String
+[`TypedArray`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray
+[`Uint8Array`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array
```

Comment:
@Trott, linking to MDN is preferable, correct?

Reply:
I don't think we've ever made that determination as a project, but it's what I tend to link to in docs, all other things being equal. But any authoritative link should be good. (It's also worth considering if the link should be something that updates or something that remains forever static. Different situations may call for different targets.)

Reply:
I originally suggested these links in https://github.com/nodejs/node/pull/32202#discussion_r426987569, but noticed that these same things are being linked to MDN in https://github.com/nodejs/node/pull/34230. At the time that I made that suggestion, it did not occur to me that MDN may be preferable. In the interest of consistency, I would like to have these all link to MDN. Does that seem reasonable?

/cc @bmeck

Reply:
MDN is more likely to be useful to the average developer. I'd certainly prefer it.